src/pages/Browse/index.jsx

```diff
@@ -8,59 +8,47 @@ import "./style.css";
 
 export default function Browse() {
   const [gameList, setGameList] = useState([]);
-  const [genreParams, setGenreParams] = useState("");
-  const options = {
-    method: "GET",
-    url: "https://rawg-video-games-database.p.rapidapi.com/games",
-    params: {
-      key: import.meta.env.VITE_RAWG_KEY,
-      page_size: "40",
-    },
-    headers: {
-      "X-RapidAPI-Key": import.meta.env.VITE_RAPID_API_KEY,
-      "X-RapidAPI-Host": "rawg-video-games-database.p.rapidapi.com",
-    },
-  };
-
-  function fetchGames(options) {
-    axios
-      .request(options)
-      .then(function (response) {
+  const axiosInstance = useAxios();
+  function fetchGames(options = {}) {
+    axiosInstance
+      .get("games", { params: { ...options } })
+      .then((response) => {
+        console.log(response);
         setGameList(response.data.results);
       })
-      .catch(function (error) {
-        console.error(error);
+      .catch((err) => {
+        console.log(err);
       });
   }
 
   useEffect(() => {
-    if (genreParams) {
-      options.params.genres = genreParams;
-    }
-    fetchGames(options);
-  }, [genreParams]);
+    fetchGames();
+  }, []);
 
   return (
     <div>
       <h1>Browse</h1>
       <div className="genres-container">
         {genreList.map(({ id, name, slug }) => {
           return (
-            <p className="genre" key={id} onClick={() => setGenreParams(slug)}>
+            <p
+              className="genre"
+              key={id}
+              onClick={() => fetchGames({ genres: slug })}
+            >
               {name}
             </p>
           );
         })}
       </div>
       <div className="browse-list">
-        {gameList.length !== 0 &&
-          gameList.map((game) => {
-            return <GameCard key={game.id} gameData={game} />;
-          })}
-        {gameList.length === 0 &&
-          Array(40)
-            .fill(1)
-            .map((item, index) => <GameCard key={index} gameData={{}} />)}
+        {gameList?.length !== 0
+          ? gameList?.map((game) => {
+              return <GameCard key={game.id} gameData={game} />;
+            })
+          : Array(40)
```

Comment:
why are we doing this ?

Reply:
I'm showing a skeleton card till the game data is fetched.

Reply:
Your code doesn't make it clear, pass a loading `true` or `showSkeleton` `true` prop or something like that so that it becomes clear that you are loading a skeleton